Resources/Private/BackendModule/src/Components/ItemsList.tsx

```diff
@@ -40,20 +44,30 @@ export default class AssetList extends PureComponent {
 
     private emptyListIndicator()
     {
-        const {assets, loading} = this.props;
+        const {items, loading} = this.props;
         return (
-            (Object.keys(assets).length === 0 && !loading) ? <span style={{backgroundColor: '#00a338', padding: '12px', fontWeight: 400, fontSize: '14px', lineHeight: 1.4, marginTop: '18px', display: 'inline-block'}}>
-                {this.translationService.translate('NEOSidekick.AiAssistant:AssetModule:listEmpty', 'There are no assets with a missing alternative text!')}
+            (Object.keys(items).length === 0 && !loading) ? <span style={{backgroundColor: '#00a338', padding: '12px', fontWeight: 400, fontSize: '14px', lineHeight: 1.4, marginTop: '18px', display: 'inline-block'}}>
+                {this.translationService.translate('NEOSidekick.AiAssistant:Module:listEmpty', 'There are no items that match the filter!')}
             </span> : null
         )
     }
 
     private renderList()
     {
-        const assets: BackendAssetModuleResultDtoInterface[] = this.assetsAsArray()
-        return (assets.map((asset: BackendAssetModuleResultDtoInterface) => (
-            <AssetListItem asset={asset} />
-        )))
+        const items: StatefulModuleItem[] = this.itemsAsArray()
+        return (items.map((item: StatefulModuleItem) => this.renderListItem(item)))
+    }
+
+    private renderListItem(item: StatefulModuleItem)
+    {
+        const {scope} = this.props;
+        switch(scope) {
+            case 'altTextGeneratorModule':
+                return <AssetListItem item={item} />;
+            case 'focusKeywordGeneratorModule':
```

Comment:
Es ist ein bisschen unschön, dass du dieses Case Statement an so vielen Stellen brauchst – insgesamt 9. Denk dran, wie wir das in Fusion machen. Nur die oberste Ebenen, und das Element selber entscheidet.

Ich vermute, das müsste nur im Root und im Item selbes ein case sein, und alles andere könnte generisch sein.